### ISSUE TITLE
[FR] Choose filter widget style based on view, not dataset

### ISSUE THREAD

**Issue by brimoor** [voxel51/fiftyone#1065]
It would be nice to update the logic in the filter sidebar to choose whether to use search-style vs explicit-checkbox-list style widgets based on the number of distinct values in the **view** rather than in the **dataset**.

In the example below, I created a view that reduced the number of distinct sample IDs from 1000+ to 10, but the `id` filter is still the search-style option, while it would be more usable if it switched to checkbox-style since the filterable set is now smaller.

<img width="1273" alt="Screen Shot 2021-06-09 at 4 47 35 PM" src="https://user-images.githubusercontent.com/25985824/121438113-71942400-c951-11eb-9de6-2b8350e550bc.png">

**Comment by benjaminpkane** [comment]
Hmm, I'm not sure this feature is makes sense. Applying matches collapses the filters to one result if there is no coincidence.

**Comment by brimoor** [comment]
Yeah I see what you mean. I think the best way to improve upon the current behavior would be if the UI for each widget were chosen based on the number of available options using the state of all **other** filters with the current filter itself omitted.

But, that would be a bit expensive to operate as it would require separate queries for each filter, so I'm fine punting on this change for now.
